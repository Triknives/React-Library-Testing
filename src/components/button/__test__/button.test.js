import React from 'react';
import ReactDOM from 'react-dom';
import Button from '../button';
import { render } from '@testing-library/react'
import '@testing-library/jest-dom';


it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<Button></Button>, div)
});

it("renders button correctly", () => {
    const {getByTestId} = render(<Button label="Click here please!"></Button>)
    expect(getByTestId('button')).toHaveTextContent("Click here please!")
})
