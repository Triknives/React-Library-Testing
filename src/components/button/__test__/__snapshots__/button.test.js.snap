// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="buttonStyle"
  data-testid="button"
>
  Save
</div>
`;
